radicle-cli/src/commands/fetch.rs

```diff
@@ -0,0 +1,149 @@
+
+use crate::terminal as term;
+use crate::terminal::args::{Args, Error, Help};
+
+pub const HELP: Help = Help {
+    name: "fetch",
+    description: "Fetch repository refs from the network",
+    version: env!("CARGO_PKG_VERSION"),
+    usage: r#"
+Usage
+
+    rad fetch <rid> [<option>...]
+
+    By default, this command will fetch from all connected seeds.
+    To instead specify a seed, use the `--seed <nid>` option.
+
+Options
+
+    --seed <nid>    Fetch seed a specific connected peer
+    --force, -f     Fetch even if the repository isn't tracked
+    --help          Print help
+
+"#,
+};
+
+#[derive(Debug)]
+pub struct Options {
+    rid: Option<Id>,
+    seed: Option<NodeId>,
+    #[allow(dead_code)]
+    force: bool,
+}
+
+impl Args for Options {
+    fn from_args(args: Vec<OsString>) -> anyhow::Result<(Self, Vec<OsString>)> {
+        use lexopt::prelude::*;
+
+        let mut parser = lexopt::Parser::from_args(args);
+        let mut rid: Option<Id> = None;
+        let mut seed: Option<NodeId> = None;
+        let mut force = false;
+
+        while let Some(arg) = parser.next()? {
+            match arg {
+                Long("force") | Short('f') => {
+                    force = true;
+                }
+                Long("seed") => {
+                    let val = parser.value()?;
+                    let val = term::args::nid(&val)?;
+                    seed = Some(val);
+                }
+                Long("help") => {
+                    return Err(Error::Help.into());
+                }
+                Value(val) if rid.is_none() => {
+                    let val = term::args::rid(&val)?;
+                    rid = Some(val);
+                }
+                _ => return Err(anyhow!(arg.unexpected())),
+            }
+        }
+
+        Ok((Options { rid, seed, force }, vec![]))
+    }
+}
+
+pub fn run(options: Options, ctx: impl term::Context) -> anyhow::Result<()> {
+    let profile = ctx.profile()?;
+    let mut node = radicle::Node::new(profile.socket());
+    let rid = match options.rid {
+        Some(rid) => rid,
+        None => {
+            let (_, rid) = radicle::rad::repo(Path::new("."))
+                .context("Current directory is not a Radicle project")?;
+
+            rid
+        }
+    };
+
+    // TODO(cloudhead): Check that we're tracking the repo, and if not, and `--force` is not
+    // used, abort with error.
+
+    let (success, failed) = if let Some(seed) = options.seed {
+        match fetch_from(rid, &seed, &mut node)? {
+            FetchResult::Success { .. } => (1, 0),
+            FetchResult::Failed { .. } => (0, 1),
+        }
+    } else {
+        fetch(rid, &mut node)?
+    };
+
+    if success == 0 {
+        term::error(format!("Failed to fetch repository from {failed} seed(s)"));
+    } else {
+        term::success!("Fetched repository from {success} seed(s)");
+    }
+    Ok(())
+}
+
+pub fn fetch(rid: Id, node: &mut Node) -> Result<(usize, usize), node::Error> {
```

Comment:
I'd suggest having a type that says which seeds succeeded or failed. That way if we want to report which seeds succeeded/failed we can. Then `usize` can be calculated using `len`.